core/keyserver/epochs.go

```diff
@@ -134,48 +136,78 @@ func (*Server) ListMutationsStream(in *pb.ListMutationsRequest, stream pb.KeyTra
 	return status.Errorf(codes.Unimplemented, "ListMutationStream is unimplemented")
 }
 
-func (s *Server) logProofs(ctx context.Context, logID, firstTreeSize int64, epoch int64) (
-	*tpb.GetLatestSignedLogRootResponse,
-	*tpb.GetConsistencyProofResponse,
-	*tpb.GetInclusionProofResponse,
-	error) {
-	// Lookup log and map info.
-	logRoot, err := s.tlog.GetLatestSignedLogRoot(ctx,
-		&tpb.GetLatestSignedLogRootRequest{
-			LogId: logID,
+// logProof holds the proof for a signed map root up to signed log root.
+type logProof struct {
+	LogRoot        *tpb.SignedLogRoot
+	LogConsistency *tpb.Proof
+	LogInclusion   *tpb.Proof
+}
+
+// logProofs returns the proofs for a given epoch.
+func (s *Server) logProofs(ctx context.Context, d *domain.Domain, firstTreeSize int64, epoch int64) (*logProof, error) {
+	logRoot, logConsistency, err := s.latestLogRootProof(ctx, d, firstTreeSize)
+	if err != nil {
+		return nil, err
+	}
+
+	// Inclusion proof.
+	secondTreeSize := logRoot.GetTreeSize()
+	logInclusion, err := s.tlog.GetInclusionProof(ctx,
+		&tpb.GetInclusionProofRequest{
+			LogId: d.LogID,
+			// SignedMapRoot must be in the log at MapRevision.
+			LeafIndex: epoch,
+			TreeSize:  secondTreeSize,
 		})
 	if err != nil {
-		glog.Errorf("tlog.GetLatestSignedLogRoot(%v): %v", logID, err)
-		return nil, nil, nil, status.Error(codes.Internal, "Cannot fetch SignedLogRoot")
+		glog.Errorf("logProofs(): log.GetInclusionProof(%v, %v, %v): %v", d.LogID, epoch, secondTreeSize, err)
+		return nil, status.Error(codes.Internal, "Cannot fetch log inclusion proof")
```

Comment:
Might want to include err in what's returned but not sure.

Reply:
done